File: Sum Of k smallest elements in BST.java

class Tree {
    int sum;
    int cnt;
    int sum(Node root, int k) { 
        sum=0;
        cnt=0;
        dfs(root,k);
        return sum;
        // Code here
    } 
    void dfs(Node root,int k){
        if(root==null)return;
        dfs(root.left,k);
        if(cnt<k){
            sum+=root.data;
            cnt++;
        }
        if(cnt<k){
            dfs(root.right,k);
        }
    }
}

File: Sum of K smallest elements  in BST.cpp
void inorder(Node* node,int &k,int &sum)
{
    if(node==NULL)
        return ;
    //if(k==0)
       //return;
   inorder(node->left,k,sum);
   if(k==0)
       return;
   sum+=node->data;
   k--;
   inorder(node->right,k,sum);
}

int sum(Node* root, int k) 
{ 
  
    // Your code here
    int sum=0;
    inorder(root,k,sum);
    return sum;
    
} 
